### PR TITLE
feat: added windsurf rules and documentation

### PR DESCRIPTION
**Pull request by pajoma**

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -0,0 +1,269 @@
...
+5. Update documentation if needed
+6. Squash and merge when approved
+
+### Review Guidelines
+
+- Check for code quality and consistency
+- Verify tests cover new functionality
+- Ensure proper error handling
+- Check for security vulnerabilities
+- Verify documentation is up to date
```
Comment (s3flsago) [inline CONTRIBUTING.md:274]: For high-impact or sensitive PRs, local checks should be done as well. E.g. Manually test basic functionalities and some edge cases, use a debugger to check, if type hints are correct.

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -0,0 +1,269 @@
...
+We follow [PEP 8](https://www.python.org/dev/peps/pep-0008/) for Python code style with the following specific rules:
+
+- **Line Length**: 120 characters 
+- **Indentation**: 4 spaces (no tabs)
+- **Quotes**: Double quotes (`"`) for strings, single quotes (`'`) for characters
+- **Imports**: Sorted and grouped in the following order:
+  1. Standard library imports
+  2. Third-party imports
+  3. Local application imports
+
```
Comment (s3flsago) [inline CONTRIBUTING.md:31]: - **Type Hints**: All new functions, methods, and class attributes must use [type hints](https://docs.python.org/3/library/typing.html). Add typing for variable definitions as well

**Comment by s3flsago** [comment]
Update the top level README.md

File: .windsurf/rules/documentation-standards.md [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+
+## 4. Naming and Structure
+
+- Use consistent, descriptive names for files, classes, and functions.
+- Organize code according to layer (e.g., controllers, services, models).
+
+## 5. Maintenance
+
+- Keep documentation up to date with any code or interface changes.
+- Outdated documentation must be updated or removed.
```
Comment (s3flsago) [inline .windsurf/rules/documentation-standards.md]: Not sure if this belongs (only) here:
## Startup Tasks and External Clients
    - External services (databases, caches, etc.) must be initialized in FastAPI's startup event.
    - Services must fail early if a required connection cannot be established.
    - All clients should support graceful shutdown via FastAPI’s shutdown event.
    - Avoid runtime initialization of clients.
    - Use dependency injection to pass connections where needed.

**Review by s3flsago — APPROVED** [review]
